Automattic/wp-calypso
# Reader: Added tracks event for time spent reading the post.

Related to https://github.com/Automattic/loop/issues/261

## Proposed Changes

* Added a tracks event to compute engaged time.
* Set readingStartTime on `componentDidMount`
* Added an addEventListener to capture tab switches

## Why are these changes being made?


* To track user engagement on the full post view of posts in the reader.

## Testing Instructions



* Open any post in the full post view in the reader by going to http://calypso.localhost:3000/read
* After the post loads, switch to a different tab. This will trigger a tracks event `calypso_reader_article_engaged_time` 
* Wait for 5 mins for the event to be captured. Visit the tracks events page to search for the tracks event and see that the property engagement_time is tracked correctly. I am not able to generate a shorthand for the link to the mc tracks page, so I am linking to the slack message  p1733991064566909-slack-C0836HR68MV which will take you to the mc tracks page for this event. 

## Pre-merge Checklist
